model/task_lifecycle.go

```diff
@@ -488,7 +488,9 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
 		if detailsCopy.Status == evergreen.TaskSucceeded && count == 0 && t.MustHaveResults {
 			detailsCopy.Status = evergreen.TaskFailed
 			detailsCopy.Description = evergreen.TaskDescriptionNoResults
-			detailsCopy.Type = evergreen.CommandTypeTest
+			if !hasFailedTests {
+				detailsCopy.Type = evergreen.CommandTypeTest
```

Comment:
I believe this ticket is about how, when there _are_ test failures, we aren't modifying the failure type (the status looks to be assigned at 474). This code block handles a different thing, where we mark things as a failure if they require test results but don't have test results. It works correctly (in this case count == 0 is the "hasFailedTests") and shouldn't be modified.